caliban/docker/build.py

```diff
@@ -583,28 +649,105 @@ def build_image(job_mode: c.JobMode,
   the problem.
 
   """
-  with ufs.TempCopy(credentials_path,
-                    tmp_name=".caliban_default_creds.json") as creds:
-    with ufs.TempCopy(adc_path, tmp_name=".caliban_adc_creds.json") as adc:
-      cache_args = ["--no-cache"] if no_cache else []
-      cmd = ["docker", "build"] + cache_args + ["--rm", "-f-", build_path]
-
-      dockerfile = _dockerfile_template(job_mode,
-                                        credentials_path=creds,
-                                        adc_path=adc,
-                                        **kwargs)
-
-      joined_cmd = " ".join(cmd)
-      logging.info("Running command: {}".format(joined_cmd))
-
-      try:
-        output, ret_code = ufs.capture_stdout(cmd, input_str=dockerfile)
-        if ret_code == 0:
-          return docker_image_id(output)
-        else:
-          error_msg = "Docker failed with error code {}.".format(ret_code)
-          raise DockerError(error_msg, cmd, ret_code)
-
-      except subprocess.CalledProcessError as e:
-        logging.error(e.output)
-        logging.error(e.stderr)
+  wrapper_path = _wrapper_script_path()
+  sql_proxy_path = _cloud_sql_proxy_wrapper_path()
+
+  temp_files = [
+      (credentials_path, CALIBAN_DEFAULT_CREDS),
+      (adc_path, CALIBAN_ADC_CREDS),
+      (wrapper_path, WRAPPER_SCRIPT),
+      (sql_proxy_path, CLOUD_SQL_WRAPPER_SCRIPT),
+  ]
+  temp_files = [ufs.TempCopy.SrcDst(src, dst) for src, dst in temp_files]
+
+  with ufs.TempCopy(files=temp_files) as tmpfiles:
+    cache_args = ["--no-cache"] if no_cache else []
+    cmd = ["docker", "build"] + cache_args + ["--rm", "-f-", build_path]
+
+    dockerfile = _dockerfile_template(
+        job_mode,
+        credentials_path=tmpfiles.get(credentials_path),
+        adc_path=tmpfiles.get(adc_path),
+        resource_files=[
+            tmpfiles.get(wrapper_path),
+            tmpfiles.get(sql_proxy_path)
+        ],
+        **kwargs,
+    )
+
+    joined_cmd = " ".join(cmd)
+    logging.info("Running command: {}".format(joined_cmd))
+
+    try:
+      output, ret_code = ufs.capture_stdout(cmd, input_str=dockerfile)
+      if ret_code == 0:
+        return docker_image_id(output)
+      else:
+        error_msg = "Docker failed with error code {}.".format(ret_code)
+        raise DockerError(error_msg, cmd, ret_code)
+
+    except subprocess.CalledProcessError as e:
+      logging.error(e.output)
+      logging.error(e.stderr)
+
+
+# ----------------------------------------------------------------------------
+def _mlflow_job_name(index: int, user: str = None) -> str:
```

Comment:
this doesn't seem like it belongs in `caliban.docker.build`... it's not related to container building. can we move to `caliban.util.mlflow`?

Reply:
ok